src/ng-select/ng-select.component.ts

```diff
@@ -486,20 +487,27 @@ export class NgSelectComponent implements OnDestroy, OnChanges, AfterViewInit, C
         }
     }
 
-    private _selectWriteValue(ngModel: any | any[]) {
-        if (!this._isDefined(ngModel)) {
+    private _handleWriteValue(ngModel: any | any[]) {
+        const isEmptyArray = ngModel && Array.isArray(ngModel) && ngModel.length === 0;
+        if (ngModel === null || isEmptyArray) {
+            this.itemsList.clearSelected();
             return;
         }
 
         const select = (val: any) => {
-            const item = this.itemsList.findItem(val);
+            let item = this.itemsList.findItem(val);
```

Comment:
`_handleWriteValue` and `_updateSelectedItems` are searching for item 2 times even if former finds it. I would suggest removing `let item = this.itemsList.findItem(val);` in the first method or marking it as `mapped` or something so `_updateSelectedItems` would search only unmapped items?